distributions/openhab/src/main/resources/bin/backup.ps1

```diff
@@ -8,103 +11,152 @@
     The directory where openHAB is installed (default: current directory).
     .PARAMETER OHBackups
     The directory to backup the files to.
-    .PARAMETER ZipFileOut
+    .PARAMETER FileName
     The name of the zip file to create
     .EXAMPLE
     Backup an openHAB instance to a zip file
     Backup-openHAB
     .EXAMPLE
     Backup the openHAB distribution in the C:\openHAB2 directory to c:\openHAB2-backup\backup.zip
-    Backup-openHAB -OHDirectory C:\openHAB2 -OHBackups c:\openHAB2-backup -ZipFileOut backup.zip
+    Backup-openHAB -OHDirectory C:\openHAB2 -OHBackups c:\openHAB2-backup -FileName backup.zip
     #>
 
     [CmdletBinding()]
     param(
-        [Parameter(ValueFromPipeline=$True)]
+        [Parameter(ValueFromPipeline = $True)]
         [string]$OHDirectory = ".",
-        [Parameter(ValueFromPipeline=$True)]
+        [Parameter(ValueFromPipeline = $True)]
         [string]$OHBackups,
-        [Parameter(ValueFromPipeline=$True)]
-        [string]$ZipFileOut
+        [Parameter(ValueFromPipeline = $True)]
+        [string]$FileName
     )
 
     begin {}
     process {
 
-        if (!([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()
-            ).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
-            throw "This script must be run as an Administrator. Start PowerShell with the Run as Administrator option"
-        }
+        Import-Module $PSScriptRoot\common.psm1 -Force
 
-        Write-Host -ForegroundColor Cyan "Checking the specified openHAB directory..."
-        if (!(Test-Path "$OHDirectory\userdata") -And !(Test-Path -Path "$OHDirectory\conf")) {
-            throw "$OHDirectory\userdata doesn't exist! Make sure you are in the " +
-                "openHAB directory or specify the -OHDirectory parameter!"
-        }
+        Write-Host ""
+        BoxMessage "openHAB 2.x.x backup script" Magenta
+        Write-Host ""
 
-        if ($OHDirectory -eq '.') {$OHDirectory = pwd }
+        CheckForAdmin
 
-        if ([Environment]::GetEnvironmentVariable("OPENHAB_CONF", "Machine")) {
-            $OHConf = [Environment]::GetEnvironmentVariable("OPENHAB_CONF", "Machine")
-        } else {
-            $OHConf = "$OHDirectory\conf"
+        Write-Host -ForegroundColor Cyan "Checking the specified openHAB directory"
+        $OHDirectory = GetOpenHABRoot $OHDirectory
+        if ($OHDirectory -eq "") {
+            return PrintAndReturn "Could not find the userdata directory! Make sure you are in the openHAB directory or specify the -OHDirectory parameter!"
         }
-        if ([Environment]::GetEnvironmentVariable("OPENHAB_USERDATA", "Machine")) {
-            $OHUserdata = [Environment]::GetEnvironmentVariable("OPENHAB_USERDATA", "Machine")
```

Comment:
I only put these checks in here because there was talk of checking for set environment variables. Is that no longer a thing?